app/services/onlinetesting/Phase1TestService.scala

```diff
@@ -268,26 +268,27 @@ trait Phase1TestService extends OnlineTestService with ResetPhase1Test {
   private def markAsInvited(application: OnlineTestApplication)
                            (newOnlineTestProfile: Phase1TestProfile): Future[Unit] = for {
     currentOnlineTestProfile <- phase1TestRepo.getTestGroup(application.applicationId)
-    updatedOnlineTestProfile = merge(currentOnlineTestProfile, newOnlineTestProfile)
-    _ <- phase1TestRepo.insertOrUpdateTestGroup(application.applicationId, updatedOnlineTestProfile)
-    _ <- phase1TestRepo.removeTestProfileProgresses(application.applicationId, determineStatusesToRemove(updatedOnlineTestProfile))
+    updatedTestProfile <- insertOrAppendNewTests(application.applicationId, currentOnlineTestProfile, newOnlineTestProfile)
+    _ <- phase1TestRepo.removeTestProfileProgresses(application.applicationId, determineStatusesToRemove(updatedTestProfile))
   } yield {
     audit("OnlineTestInvited", application.userId)
   }
 
-  private def merge(currentProfile: Option[Phase1TestProfile], newProfile: Phase1TestProfile): Phase1TestProfile = currentProfile match {
-    case None =>
-      newProfile
-    case Some(profile) =>
-      val scheduleIdsToArchive = newProfile.tests.map(_.scheduleId)
-      val existingTestsAfterUpdate = profile.tests.map(t =>
-        if (scheduleIdsToArchive.contains(t.scheduleId)) {
-          t.copy(usedForResults = false)
-        } else {
-          t
+  private def insertOrAppendNewTests(applicationId: String, currentProfile: Option[Phase1TestProfile],
+                    newProfile: Phase1TestProfile): Future[Phase1TestProfile] = {
+    (currentProfile match {
+      case None => phase1TestRepo.insertOrUpdateTestGroup(applicationId, newProfile)
+      case Some(profile) =>
+        val scheduleIdsToArchive = newProfile.tests.map(_.scheduleId)
+        val inactiveTests = profile.tests.filter(t => scheduleIdsToArchive.contains(t.scheduleId)).map(_.cubiksUserId)
+        Future.traverse(inactiveTests)(phase1TestRepo.markTestAsInactive).flatMap { _ =>
```

Comment:
Some of these tests could already be marked as inactive, should we filter these out so we can avoid making repo calls when we don't need to?